lib/modules/manager/cpanfile/extract.ts

```diff
@@ -0,0 +1,32 @@
+import { logger } from '../../../logger';
+import type { PackageFileContent } from '../types';
+import { cpanfile } from './language';
+import { Ctx, query } from './parser';
+
+export function extractPackageFile(
+  content: string,
+  packageFile?: string
+): PackageFileContent | null {
+  let parsedResult: Ctx | null = null;
+
+  try {
+    parsedResult = cpanfile.query(content, query, {
+      deps: [],
+    });
+  } catch (err) /* istanbul ignore next */ {
```

Comment:
can you add a test instead?

Reply:
How do I let `Language.query` throw an error?